File: Java/Vetores/alturas.java
// Exercício: Alturas
// Objetivo: armazenar dados de pessoas, calcular a altura média e o percentual de pessoas com menos de 16 anos.

import java.util.Locale;
import java.util.Scanner;

public class alturas {
    public static void main(String[] args) {
       
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        int n, cont;
        double media, soma, percentual;

        System.out.print("Quantas pessoas serao digitadas? ");
        n = sc.nextInt();

        String[] nome = new String[n];
        int[] idade = new int[n];
        double[] altura = new double[n];

        for (int i = 0; i < n; i++) {
            System.out.println("Dados da " + (i + 1) + "a pessoa: ");
            System.out.print("Nome: ");
            nome[i] = sc.next();
            System.out.print("Idade: ");
            idade[i] = sc.nextInt();
            System.out.print("Altura: ");
            altura[i] = sc.nextDouble(); 
        }

        System.out.println();

        soma = 0;
        for (int i = 0; i < n; i++) {
            soma = soma + altura[i];
        }

        media = 0;
        media = soma / n;
        System.out.println("Altura média: " + String.format("%.2f", media));

        cont = 0;
        for (int i = 0; i < n; i++) {
            if (idade[i] < 16) {
                cont = cont + 1;
            }
        }

        percentual = (double) cont * 100 / n;

        System.out.println("Pessoas com menos de 16 anos: " + String.format("%.1f", percentual) + "%");

        for (int i = 0; i < n; i++) {
            if (idade[i] < 16) {
                System.out.println(nome[i]);
            }
        }

        sc.close();
    }
}

File: Python/vetores/alturas.py
# Exercício: Alturas
# Objetivo: armazenar nome, idade e altura das pessoas, calcular a altura média e o percentual de menores de 16 anos.

n = int(input("Quantas pessoas serao digitadas? "))

nome = [0 for x in range(n)]
idade = [0 for x in range(n)]
altura = [0.0 for x in range(n)]

for i in range(n):
    print("Dados da", i+1 ,"a pessoa: ")
    nome[i] = input("Nome: ")
    idade[i] = int(input("Idade: "))
    altura[i] = float(input("Altura: "))

soma = 0
for i in range(n):
    soma = soma + altura[i]

media = 0
media = soma / n
print(f"Altura média: {media:.2f}")

cont = 0
for i in range(n):
    if idade[i] < 16:
        cont = cont + 1

percentual = cont * 100 / n

print(f"Pessoas com menos de 16 anos: {percentual:.1f}%")

for i in range(n):
    if idade[i] < 16:
        print(nome[i])